celery/bin/worker.py

```diff
@@ -131,7 +169,6 @@ def detach(path, argv, logfile=None, pidfile=None, uid=None,
               '--statedb',
               cls=CeleryOption,
               type=click.Path(),
-              callback=lambda ctx, _, value: value or ctx.obj.app.conf.worker_state_db,
```

Comment:
This cannot be removed. The default value must be derived from the app if it is not supplied.

Reply:
It's not removed, only moved below, so the app will not be loaded too soon. See the call to worker.start in this function

Reply:
Let's address the other issues first and see if we can find the way to write this code as Click intended it to be written.